oss-internship-2020/libtiff/example/main_sandboxed.cc

```diff
@@ -0,0 +1,305 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <array>
+#include <cstdint>
+#include <cstdlib>
+#include <cstring>
+
+#include "../sandboxed.h"
+#include "sandboxed_api/sandbox2/util/fileops.h"
+#include "sandboxed_api/sandbox2/util/path.h"
+#include "sandboxed_api/vars.h"
+#include "tiffio.h"  // NOLINT(build/include)
+
+// sapi functions:
+//  TIFFTileSize
+//  TIFFOpen
+//  TIFFReadEncodedTile
+//  TIFFSetField
+//  TIFFClose
+//  TIFFReadRGBATile
+//  TIFFGetField
+
+constexpr std::array<uint8_t, 6> kCluster0 = {0, 0, 2, 0, 138, 139};
+constexpr std::array<uint8_t, 6> kCluster64 = {0, 0, 9, 6, 134, 119};
+constexpr std::array<uint8_t, 6> kCluster128 = {44, 40, 63, 59, 230, 95};
+
+namespace {
+
+absl::Status CheckCluster(int cluster, const sapi::v::Array<uint8_t>& buffer,
+                          const std::array<uint8_t, 6>& expected_cluster) {
+  uint8_t* target = buffer.GetData() + cluster * 6;
+
+  if (!std::memcmp(target, expected_cluster.data(), 6)) {
+    return absl::OkStatus();
+  }
+
+  return absl::InternalError(
+      absl::StrCat("Cluster ", cluster, " did not match expected results.\n",
+                   "Expect: ", expected_cluster[0], "\t", expected_cluster[1],
+                   "\t", expected_cluster[4], "\t", expected_cluster[5], "\t",
+                   expected_cluster[2], "\t", expected_cluster[3], "\n"));
+}
+
+absl::Status CheckRgbPixel(int pixel, int min_red, int max_red, int min_green,
+                           int max_green, int min_blue, int max_blue,
+                           const sapi::v::Array<uint8_t>& buffer) {
+  uint8_t* rgb = buffer.GetData() + 3 * pixel;
+
+  if (rgb[0] >= min_red && rgb[0] <= max_red && rgb[1] >= min_green &&
+      rgb[1] <= max_green && rgb[2] >= min_blue && rgb[2] <= max_blue) {
+    return absl::OkStatus();
+  }
+
+  return absl::InternalError(absl::StrCat(
+      "Pixel ", pixel, " did not match expected results.\n", "Got R=", rgb[0],
```

Comment:
a..b usually describes the range from a till b - 1. I'd go with `..=` (rust syntax), but no need to change this if you don't think it's worth it.